File: packages/sdk/sdk-ts/examples/embeddings.ts
import { Phaseo } from "../src";

const key = process.env.PHASEO_API_KEY;
if (!key) throw new Error("Set PHASEO_API_KEY");

async function main() {
  const client = new Phaseo({ apiKey: key });
  const response = await client.generateEmbedding({
    model: "google/gemini-embedding-001",
    input: "Vector search uses embeddings to compare meaning.",
  });
  console.log("embedding length:", response.data?.[0]?.embedding?.length);
}

main().catch((err) => {
  console.error(err);
  process.exitCode = 1;
});

File: packages/sdk/sdk-py/examples/embeddings.py
import os
from phaseo import Phaseo

api_key = os.getenv("PHASEO_API_KEY")
if not api_key:
	raise RuntimeError("Set PHASEO_API_KEY")

client = Phaseo(api_key=api_key)

response = client.generate_embedding({
	"model": "google/gemini-embedding-001",
	"input": "Vector search uses embeddings to compare meaning.",
})

print("embedding length:", len(response["data"][0]["embedding"]))
